src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/config/RetryAspect.java

```diff
@@ -0,0 +1,56 @@
+package ca.bc.gov.open.jag.tco.oracledataapi.config;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Aspect
+@Component
+//@EnableRetry
+public class RetryAspect {
+
+	private static Logger logger = LoggerFactory.getLogger(RetryAspect.class);
+
+	@Value("${ords.api.retry.count}")
+	private int maxAttempts;
+
+	@Value("${ords.api.retry.delay}")
+	private long retryDelay;
+
+	// This pointcut will match any public method on any class in the ords repository package (DisputeRepositoryImpl, JJDisputeRepositoryImpl, etc.)
+	// as well as the LookupValuesApi class that retrieves lookup values from ords.
+	@Pointcut("execution(public * ca.bc.gov.open.jag.tco.oracledataapi.repository.impl.ords.*.*(..)) || "
+			+ "execution(public * ca.bc.gov.open.jag.tco.oracledataapi.ords.occam.api.LookupValuesApi.*(..))")
+	public void retryOnOrdsApiMethods() {
+	}
+
+	//Unfortunately, @Retryable is not supported in AspectJ so this doesn't work
+	//@Retryable(maxAttemptsExpression = "${ords.api.retry.count}", backoff = @Backoff(delayExpression = "${ords.api.retry.delay}"))
+	@Around("retryOnOrdsApiMethods()")
+	public Object around(ProceedingJoinPoint pjp) throws Throwable {
+		for (int i = 0; i < maxAttempts; i++) {
+			try {
+				return pjp.proceed();
+			} catch (Exception e) {

```

Comment:
I wonder if we can interrogate this exception and add some smarts to it.  Are there known exception types we may not want to retry on.

Reply:
Yes, it's possible. I can dig into the logs to see what exceptions have been thrown.